core/src/test/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -0,0 +1,117 @@
+                                                                                      MappedFieldType fieldType,
+                                                                                      IndexSearcher indexSearcher) throws IOException {
+        IndexSettings indexSettings = new IndexSettings(
+            IndexMetaData.builder("_index").settings(Settings.builder().put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT))
+                .numberOfShards(1)
+                .numberOfReplicas(0)
+                .creationDate(System.currentTimeMillis())
+                .build(),
+            Settings.EMPTY
+        );
+
+        Engine.Searcher searcher = new Engine.Searcher("aggregator_test", indexSearcher);
+        QueryCache queryCache = new DisabledQueryCache(indexSettings);
+        QueryCachingPolicy queryCachingPolicy = new QueryCachingPolicy() {
+            @Override
+            public void onUse(Query query) {
+            }
+
+            @Override
+            public boolean shouldCache(Query query) throws IOException {
+                // never cache a query
+                return false;
+            }
+        };
+        ContextIndexSearcher contextIndexSearcher = new ContextIndexSearcher(searcher, queryCache, queryCachingPolicy);
+
+        CircuitBreakerService circuitBreakerService = new NoneCircuitBreakerService();
+        SearchContext searchContext = mock(SearchContext.class);
+        when(searchContext.searcher()).thenReturn(contextIndexSearcher);
+        when(searchContext.parseFieldMatcher()).thenReturn(ParseFieldMatcher.STRICT);
+        when(searchContext.bigArrays()).thenReturn(new MockBigArrays(Settings.EMPTY, circuitBreakerService));
+        when(searchContext.fetchPhase())
+            .thenReturn(new FetchPhase(Arrays.asList(new FetchSourceSubPhase(), new DocValueFieldsFetchSubPhase())));
+
+        // TODO: now just needed for top_hits, this will need to be revised for other agg unit tests:
+        MapperService mapperService = mock(MapperService.class);
+        when(mapperService.hasNested()).thenReturn(false);
+        when(searchContext.mapperService()).thenReturn(mapperService);
+
+        SearchLookup searchLookup = new SearchLookup(mapperService, mock(IndexFieldDataService.class), new String[]{"type"});
+        when(searchContext.lookup()).thenReturn(searchLookup);
+
+        QueryShardContext queryShardContext = mock(QueryShardContext.class);
+        IndexFieldData<?> fieldData = fieldType.fielddataBuilder().build(indexSettings, fieldType, null, circuitBreakerService,
+                mock(MapperService.class));
+        when(queryShardContext.fieldMapper(anyString())).thenReturn(fieldType);
+        when(queryShardContext.getForField(any())).thenReturn(fieldData);
+        when(searchContext.getQueryShardContext()).thenReturn(queryShardContext);
+
+        @SuppressWarnings("unchecked")
+        A aggregator = (A) aggregatorBuilder.build(searchContext, null).create(null, true);
```

Comment:
I don't think the generics on this method are buying anything since we do a blind cast here?